dotnet/src/SemanticKernel.Abstractions/Search/TextSearchDelegates.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.Search;
+
+/// <summary>
+/// Delegate to map a search result instance from an arbitrary <see cref="object"/> to a <see cref="string"/>
+/// </summary>
+/// <param name="result">The search result to be mapped.</param>
+public delegate string MapSearchResultToString(object result);
+
+/// <summary>
+/// Delegate to map a search result instance from an arbitrary <see cref="object"/> to a <see cref="TextSearchResult"/>
+/// </summary>
+/// <param name="result">The search result to be mapped.</param>
+public delegate TextSearchResult MapSearchResultToTextSearchResult(object result);
```

Comment:
Just wondering, have you considered using interfaces instead of delegates and pros/cons of each approach? In memory design we use an interface for mapping purposes, and it works good so far:
https://github.com/microsoft/semantic-kernel/blob/21507472dbfb2f82c4cee14827624e2d14dc0b39/dotnet/src/SemanticKernel.Abstractions/Data/IVectorStoreRecordMapper.cs#L13

Reply:
Will follow the same pattern, tracking the change here: https://github.com/microsoft/semantic-kernel/issues/8390